### PR TITLE
Move bot to postgres

### PR DESCRIPTION
**Pull request by twof**
Making this move because we'd like to be able to preserve test results so that we can do fun things like track performance metrics over time.

### Checklist

- [ ] Circle CI is passing (code compiles and passes tests).
- [ ] There are no breaking changes to public API.
- [ ] New test cases have been added where appropriate.
- [ ] All new code has been commented with doc blocks `///`.

### CODE REVIEW

File: Sources/App/configure.swift [hunk 1]
```diff
@@ -1,23 +1,26 @@
...
     // Register providers first
-    try services.register(FluentSQLiteProvider())
+    try services.register(FluentPostgreSQLProvider())
 
     // Register routes to the router
     let router = EngineRouter.default()
     try routes(router)
     services.register(router, as: Router.self)
     
+    Environment.dotenv(filename: ".env")
```
Comment (twof) [inline Sources/App/configure.swift:15]: Random QOL update. Shouldn't have an impact on this PR

File: Sources/App/configure.swift [hunk 2]
```diff
@@ -32,17 +35,31 @@ public func configure(_ config: inout Config, _ env: inout Environment, _ servic
...
     // Configure migrations
     var migrations = MigrationConfig()
-    migrations.prepareCache(for: .sqlite)
-    migrations.add(model: PerformanceTestResults.self, database: .sqlite)
+    migrations.prepareCache(for: .psql)
+    migrations.add(model: PerformanceTestResults.self, database: .psql)
     services.register(migrations)
 }
+
+fileprivate func dbConfig(environment: Environment) throws -> PostgreSQLDatabaseConfig {
```
Comment (twof) [inline Sources/App/configure.swift:54]: Adapted from vapor cloud v2 docs

**Review by MrLotU — APPROVED** [review]

File: Sources/App/configure.swift (hunk 2 above)
Comment (MrLotU) [reply Sources/App/configure.swift:54]: Actually, could you make this more like the Vapor-TIL book setup? https://github.com/raywenderlich/vapor-til/blob/master/Sources/App/configure.swift#L58 leaves some more flexibility for running locally (I prefer not to use my postgres user for things for example)

Comment (twof) [reply Sources/App/configure.swift:54]: Updated. Let me know what you think. There were some parts of the vapor-til config func that I didn't like so I opted to update them.

File: Sources/App/configure.swift [hunk 3]
```diff
@@ -50,16 +52,41 @@ public func configure(_ config: inout Config, _ env: inout Environment, _ servic
...
+    let password: String? = Environment.get("DATABASE_PASSWORD") ?? nil
+    let databaseName: String?
+    let databasePort: Int
+    
     switch environment {
-    case .development:
-        return PostgreSQLDatabaseConfig.init(hostname: "localhost", port: 5433, username: "postgres")
-    default:
+    case .testing:
+        databasePort = Environment.get("DATABASE_PORT") ?? 5433
```
Comment (MrLotU) [inline Sources/App/configure.swift]: Why is this not the default port? 🤔

File: Sources/App/configure.swift [hunk 4]
```diff
@@ -50,16 +52,41 @@ public func configure(_ config: inout Config, _ env: inout Environment, _ servic
...
 
 fileprivate func dbConfig(environment: Environment) throws -> PostgreSQLDatabaseConfig {
+    // Configure a database
+    let databaseConfig: PostgreSQLDatabaseConfig
+    let hostname = Environment.get("DATABASE_HOSTNAME") ?? "localhost"
+    let username = Environment.get("DATABASE_USER") ?? "vapor"
+    let password: String? = Environment.get("DATABASE_PASSWORD") ?? nil
+    let databaseName: String?
+    let databasePort: Int
+
```
Comment (MrLotU) [inline Sources/App/configure.swift:64]: Could you add a "catchall" for a `DB_URL` env var? I personally prefer using 1 env var for my DB Config

File: Sources/App/configure.swift (hunk 3 above)
Comment (twof) [reply Sources/App/configure.swift]: updated

File: Sources/App/configure.swift (hunk 4 above)
Comment (twof) [reply Sources/App/configure.swift:64]: Done https://github.com/vapor/bot-github/pull/11/files#diff-cc9aa3d707deffc5f5cb05865ea525c7R65

Comment (MrLotU) [reply Sources/App/configure.swift:64]: Awesome!

**Comment by penny-coin** [comment]
Hey @twof, you just merged a pull request, have a coin! 

You now have 60 coins.

**Pull request merged**